src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,44 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate localDateFrom;
+        LocalDate localDateTo;
+        StringBuilder builder = new StringBuilder();
+
+        try {
+            localDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+            localDateTo = LocalDate.parse(dateTo, FORMATTER);
+        } catch (DateTimeParseException e) {          
+            throw new RuntimeException("Input date has incorrect format");
+        }
+        builder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        try {
+            for (String name : names) {
+                int salary = 0;
+    
+                for (String record : data) {
+                    String[] dataLineArray = record.split(" ");               
+                    LocalDate dataBaseDate = LocalDate.parse(dataLineArray[0], FORMATTER);
+    
+                    if (dataBaseDate.compareTo(localDateFrom) >= 0 
+                            && dataBaseDate.compareTo(localDateTo) <= 0 
+                            && dataLineArray[1].equals(name)) {
+                        salary += Integer.parseInt(dataLineArray[2]) 
+                                * Integer.parseInt(dataLineArray[3]);
```

Comment:
Create constants for indexes.
![image](https://user-images.githubusercontent.com/20326858/203299867-b086768e-3e9b-4298-b2c2-9f3698c262a6.png)